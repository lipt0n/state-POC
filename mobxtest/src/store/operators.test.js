import o from './operators'
import operators_response from '../../../server/operators'
beforeEach(() => {
    fetch.resetMocks()
  })
test('operators',()=>{
    const oi = new o({})
    expect(oi.operators).toMatchSnapshot()
})
test('fetch operators',async ()=>{
    fetch.mockResponse(JSON.stringify(operators_response))
    const oi = new o({})
    const fetchMyData = await oi.fetchProjects()
    expect(oi.state).toEqual("done")  
})
test('fetch operators fail',async ()=>{
    fetch.mockReject(new Error('fake error message'))
    const oi = new o({})
    const fetchMyData = await oi.fetchProjects()
    expect(oi.state).toEqual("error")   
})