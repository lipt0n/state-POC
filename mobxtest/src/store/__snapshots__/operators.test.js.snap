// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`operators 1`] = `Array []`;
